doc/proxy-clients.rst

```diff
@@ -1,15 +1,71 @@
 Caching Proxy Clients
 =====================
 
-This library ships with clients for the Varnish, NGINX and Symfony built-in caching proxies. You
-can use the clients either wrapped by the :doc:`cache invalidator <cache-invalidator>`
-(recommended), or directly for low-level access to invalidation functionality.
+This library ships with clients for the Varnish and NGINX caching servers and
+the Symfony built-in HTTP cache. You can use the clients either wrapped by the
+:doc:`cache invalidator <cache-invalidator>` (recommended), or directly for
+low-level access to invalidation functionality. Which client you need depends on
+which caching solution you use.
 
 .. _client setup:
 
 Setup
 -----
 
+Install HTTP Adapter
+~~~~~~~~~~~~~~~~~~~~
+
+Because the clients send invalidation requests over HTTP, an `HTTP adapter`_
+must be installed. Which one you need depends on the HTTP client library that
+you use in your project. For instance, if you use Guzzle 6 in your project,
+install the appropriate adapter:
+
+.. code-block:: bash
+
+    $ composer require php-http/guzzle6-adapter
+
+You also need a `PSR-7 message implementation`_. If you use Guzzle 6, Guzzle’s
+implementation is already included. If you use another client, install one of
+the implementations. Recommended:
+
+.. code-block:: bash
+
+    $ composer require guzzlehttp/psr7
+
+Alternatively:
+
+.. code-block:: bash
+
+    $ composer require zendframework/zend-diactoros
+
+.. _custom HTTP client:
```

Comment:
could we say "configure HTTP adapter" then? i guess doing this also avoids the autodiscovery, in case there are multiple adapters in the codebase and i want to force a specific one.